tests/test_io.py

```diff
@@ -0,0 +1,426 @@
+        )
+        assert isinstance(fixed_mol, Residue)
+
+        # Test with a Molecule object (using SMILES template)
+        fixed_mol_custom = ProteinHelper.fix_molecule_bond_orders(
+            INPUT_MOL.residues[1],
+            templates=[
+                {
+                    "TPO": {
+                        "name": "TPO",
+                        "SMILES": "C[C@H]([C@@H](C(=O))N)OP(=O)(O)O",
+                    }
+                }
+            ],
+        )
+        assert isinstance(fixed_mol_custom, Residue)
+
+        # check two residue are equal and bond orders are fixed
+        for at1, at2 in zip(
+            fixed_mol.GetAtoms(), fixed_mol_custom.GetAtoms(), strict=False
+        ):
+            assert at1.GetSymbol() == at2.GetSymbol()
+
+        bond1_info, bond2_info = [], []
+        for bond in fixed_mol.GetBonds():
+            if bond.GetBeginAtomIdx() < bond.GetEndAtomIdx():
+                bond1_info.append(
+                    f"{bond.GetBeginAtomIdx()}_{bond.GetEndAtomIdx()}_"
+                    f"{int(bond.GetBondTypeAsDouble())}"
+                )
+            else:
+                bond1_info.append(
+                    f"{bond.GetEndAtomIdx()}_{bond.GetBeginAtomIdx()}_"
+                    f"{int(bond.GetBondTypeAsDouble())}"
+                )
+
+        for bond in fixed_mol_custom.GetBonds():
+            if bond.GetBeginAtomIdx() < bond.GetEndAtomIdx():
+                bond2_info.append(
+                    f"{bond.GetBeginAtomIdx()}_{bond.GetEndAtomIdx()}_"
+                    f"{int(bond.GetBondTypeAsDouble())}"
+                )
+            else:
+                bond2_info.append(
+                    f"{bond.GetEndAtomIdx()}_{bond.GetBeginAtomIdx()}_"
+                    f"{int(bond.GetBondTypeAsDouble())}"
+                )
+        bond1_info = sorted(bond1_info)
+        bond2_info = sorted(bond2_info)
+        assert_equal(bond1_info, bond2_info)
+
+        # Test: not found template for residue (TPO) in default templates
+        with pytest.raises(
+            ValueError,
+            match=r"Failed to find template for residue: \'TPO\'",
+        ):
+            ProteinHelper.fix_molecule_bond_orders(INPUT_MOL.residues[1])
+
+    def test_forcefield_guesser(self) -> None:
+        """Test the forcefield guesser."""
+        from prolif.io.protein_helper import ProteinHelper
+
+        # Test with a known forcefield
+        forcefield = ProteinHelper.forcefield_guesser({"HSD"})
+        assert forcefield == "charmm"
+
+        forcefield = ProteinHelper.forcefield_guesser({"NASP"})
+        assert forcefield == "amber"
+
+        forcefield = ProteinHelper.forcefield_guesser({"ASN1"})
+        assert forcefield == "gromos"
+
+        forcefield = ProteinHelper.forcefield_guesser({"HISD"})
+        assert forcefield == "oplsaa"
+
+        forcefield = ProteinHelper.forcefield_guesser({"CYS"})
+        assert forcefield == "unknown"
+
+    def test_standardize_protein(
```

Comment:
this test does too many different things. Tests should be written as functions and have a single responsibility e.g. one test for checking if a warning is emitted, one test for checking if the output is as expected...etc.